File: src/main/java/com/arangodb/intellij/aql/templates/AqlTemplateContext.java
package com.arangodb.intellij.aql.templates;

import org.jetbrains.annotations.NotNull;

import com.arangodb.intellij.aql.fileTypes.AqlFileType;
import com.intellij.codeInsight.template.TemplateActionContext;
import com.intellij.codeInsight.template.TemplateContextType;

public class AqlTemplateContext extends TemplateContextType {
    protected AqlTemplateContext() {
        super("AQL", "AQL");
    }

    @Override
    public boolean isInContext(@NotNull final TemplateActionContext templateActionContext) {
        return templateActionContext.getFile().getFileType().equals(AqlFileType.INSTANCE);
    }
}

File: src/main/kotlin/com/arangodb/intellij/aql/templates/AqlTemplateContext.kt
package com.arangodb.intellij.aql.templates

import com.arangodb.intellij.aql.fileTypes.AqlFileType
import com.intellij.codeInsight.template.TemplateActionContext
import com.intellij.codeInsight.template.TemplateContextType

class AqlTemplateContext : TemplateContextType("AQL", "AQL") {

    override fun isInContext(templateActionContext: TemplateActionContext): Boolean =
        templateActionContext.file.fileType == AqlFileType
}
